sklearn/metrics/tests/test_classification.py

```diff
@@ -2804,3 +2807,27 @@ def test_classification_metric_pos_label_types(metric, classes):
         y_pred = y_true.copy()
     result = metric(y_true, y_pred, pos_label=pos_label)
     assert not np.any(np.isnan(result))
+
+
+@pytest.mark.parametrize(
+    "scoring",
+    [
+        make_scorer(f1_score, zero_division=np.nan),
+        make_scorer(fbeta_score, beta=2, zero_division=np.nan),
+        make_scorer(precision_score, zero_division=np.nan),
+        make_scorer(recall_score, zero_division=np.nan),
+    ],
+)
+def test_classification_metric_division_by_zero_nan_validaton(scoring):
+    """Check that we validate `np.nan` properly for classification metrics.
+
+    With `n_jobs=2` in cross-validation, the `np.nan` used for the singleton will be
+    different in the sub-process and we should not use the `is` operator but
+    `math.isnan`.
+
+    Non-regression test for:
+    https://github.com/scikit-learn/scikit-learn/issues/27563
```

Comment:
When i run this test without the other changes from this PR it passes. Does it fail for you?

Seems like there is an ingredient missing to make it reliably(?) reproduce the problem :-/

Reply:
Uhm I need to have a second look then. The following code was failing for me: https://github.com/scikit-learn/scikit-learn/issues/27563#issuecomment-1758332192

Reply:
Oh no I see, I forgot to put `error_score='raise'` in the `cross_val_score`.

Reply:
Now the test fails for me :)